### PR TITLE
fix(interactive): fix the bug in filtering after intersection in GIE Runtime in distributed computation

### LINKED ISSUE #3360: [BUG] unexpected results of filtering after intersection in GIE Runtime in distributed computation
**Describe the bug**

When starting GIE with 2 servers, the following query outputs unexpected results (on LDBC graph):

```
Match (comment:COMMENT|POST)-[:HASCREATOR]->(person:PERSON),
              (comment:COMMENT|POST)-[:HASTAG]->(tag:TAG),
              (person:PERSON) -[:HASINTEREST]->(tag:TAG)
        Return count(person);
```

The bug occurs when we do some filtering on the intersected vertices in the pattern.

### PR DESCRIPTION
**Pull request by BingqingLyu**
## What do these changes do?

This pr:
1) fix the bug in filtering after intersection in distributed computation;
2) optimize the case of filtering by label only that doesn't necessary to query the storage, in Auxilia operator.

## Related issue number

Fixes #3360

### CODE REVIEW

File: interactive_engine/executor/ir/graph_proxy/src/apis/graph/mod.rs [hunk 1]
```diff
@@ -174,4 +174,12 @@ impl QueryParams {
             None
         }
     }
+
+    // is_queryable doesn't consider tables as we assume that the table info can be inferred directly from current data.
+    pub fn is_queryable(&self) -> bool {
+        !(self.filter.is_none()
+            && self.limit.is_none()
```
Comment (longbinlai) [inline interactive_engine/executor/ir/graph_proxy/src/apis/graph/mod.rs:181]: Why we need to consider limit and sample_ratio then. And if this is the case, directly using `! ( self.filter.is_none() && self.columns.is_none())` without introducing this arbitrary function is a better idea.

Comment (BingqingLyu) [reply interactive_engine/executor/ir/graph_proxy/src/apis/graph/mod.rs]: Done

File: flex/engines/hqps_db/database/adj_list.h [hunk 2]
```diff
@@ -114,12 +114,10 @@ class SinglePropGetter {
   using value_type = T;
   static constexpr size_t prop_num = 1;
   SinglePropGetter() {}
-  SinglePropGetter(std::shared_ptr<TypedRefColumn<T>> c) : column(c) {
-    CHECK(column.get() != nullptr);
-  }
+  SinglePropGetter(std::shared_ptr<TypedRefColumn<T>> c) : column(c) {}
 
   inline value_type get_view(vid_t vid) const {
-    if (vid == NONE) {
```
Comment (longbinlai) [inline flex/engines/hqps_db/database/adj_list.h:122]: CAN use columns.is_nullptr()

File: interactive_engine/executor/ir/runtime/src/process/operator/map/get_v.rs [hunk 3]
```diff
@@ -159,17 +159,22 @@ impl FilterMapFunction<Record, Record> for AuxiliaOperator {
             // e.g., for g.V().out().as("a").has("name", "marko"), we should compile as:
             // g.V().out().auxilia(as("a"))... where we give alias in auxilia,
             //     then we set tag=None and alias="a" in auxilia
-            // 1. filter by labels.
+
+            // 1. If to filter by labels, and the entry itself carries label information already, directly eval it without query the store
```
Comment (longbinlai) [inline interactive_engine/executor/ir/runtime/src/process/operator/map/get_v.rs:163]: Can the following code be replaced with params.ha_labels()?

File: interactive_engine/executor/ir/runtime/src/process/operator/map/get_v.rs [hunk 4]
```diff
@@ -159,17 +159,22 @@ impl FilterMapFunction<Record, Record> for AuxiliaOperator {
...
             if !self.query_params.labels.is_empty() && entry.label().is_some() {
                 if !self
                     .query_params
                     .labels
                     .contains(&entry.label().unwrap())
                 {
+                    // pruning by labels
                     return Ok(None);
+                } else if self.query_params.filter.is_none() && self.query_params.columns.is_none() {
```
Comment (longbinlai) [inline interactive_engine/executor/ir/runtime/src/process/operator/map/get_v.rs:172]: Use has_predicates & has_columns?

File: interactive_engine/executor/ir/core/src/plan/logical.rs [hunk 5]
```diff
@@ -1477,7 +1477,13 @@ fn is_whole_graph(operator: &pb::logical_plan::Operator) -> bool {
                     && scan
                         .params
                         .as_ref()
-                        .map(|params| !params.is_queryable() && is_params_all_labels(params))
+                        .map(|params| {
+                            !(params.has_columns()
+                                || params.has_predicates()
+                                || params.has_sample()
+                                || params.has_limit())
+                                && is_params_all_labels(params)
```
Comment (longbinlai) [inline interactive_engine/executor/ir/core/src/plan/logical.rs:1485]: Can is_params_all_labels be replaced with params.has_labels()?

File: interactive_engine/executor/ir/runtime/src/process/operator/map/get_v.rs (hunk 3 above)
Comment (BingqingLyu) [reply interactive_engine/executor/ir/runtime/src/process/operator/map/get_v.rs:163]: Done.

File: interactive_engine/executor/ir/runtime/src/process/operator/map/get_v.rs (hunk 4 above)
Comment (BingqingLyu) [reply interactive_engine/executor/ir/runtime/src/process/operator/map/get_v.rs]: Done

File: interactive_engine/executor/ir/core/src/plan/logical.rs (hunk 5 above)
Comment (BingqingLyu) [reply interactive_engine/executor/ir/core/src/plan/logical.rs:1485]: Maybe not. Here we have to confirm it is the "whole graph", that it either specifies all labels, or no labels (also indicates that all labels is allowed).

**Review by longbinlai — APPROVED** [review]
LGTM

**Pull request merged**